projects/wave-core-new/src/lib/colors/color-breakpoint-input/color-breakpoint-input.component.ts

```diff
@@ -0,0 +1,103 @@
+import {
+    Component,
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    AfterViewInit,
+    Input,
+    forwardRef,
+    OnChanges,
+    SimpleChanges,
+} from '@angular/core';
+
+import {ControlValueAccessor, NG_VALUE_ACCESSOR} from '@angular/forms';
+import {ColorBreakpoint} from '../color-breakpoint.model';
+import {Color, stringToRgbaStruct} from '../color';
+
+@Component({
+    selector: 'wave-color-breakpoint',
+    templateUrl: './color-breakpoint.component.html',
+    styleUrls: ['./color-breakpoint.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [{provide: NG_VALUE_ACCESSOR, useExisting: forwardRef(() => ColorBreakpointInputComponent), multi: true}],
+})
+export class ColorBreakpointInputComponent implements ControlValueAccessor, AfterViewInit, OnChanges {
+    @Input() editAttribute: false;
+    @Input() editColor: false;
+    @Input() inputType: 'number' | 'string' = 'number';
+    @Input() attributePlaceholder = 'attribute';
+    @Input() colorPlaceholder: 'color';
+
+    onTouched: () => void;
+    onChange: (_: ColorBreakpoint) => void = undefined;
+
+    private _colorBreakpoint: ColorBreakpoint;
+
+    constructor(private changeDetectorRef: ChangeDetectorRef) {}
+
+    get colorBreakpoint(): ColorBreakpoint {
+        return this._colorBreakpoint;
+    }
+
+    // set accessor including call the onchange callback
+    set colorBreakpoint(breakpoint: ColorBreakpoint) {
+        if (breakpoint && !breakpoint.equals(this._colorBreakpoint)) {
+            this._colorBreakpoint = breakpoint;
+        }
+    }
+
+    updateValue(value: number | string): void {
+        if (value && value !== this.colorBreakpoint.value) {
+            if (this.inputType === 'number' && typeof value === 'string') {
+                this.colorBreakpoint.setValue(parseFloat(value as string));
+            } else {
+                this.colorBreakpoint.setValue(value);
+            }
+            this.propagateChange();
+        }
+    }
+
+    updateColor(color: string): void {
+        // TODO: should this really clone?
+        if (color) {
+            const clr = Color.fromRgbaLike(stringToRgbaStruct(color));
+            if (!clr.equals(this._colorBreakpoint.rgba)) {
```

Comment:
there is no rgba in ColorBreakpoint

Reply:
You are right, I fixed it.
795bc324f66ff98dc00435d0a506a018f6e340ea